managed/src/main/java/com/yugabyte/yw/cloud/aws/AWSCloudImpl.java

```diff
@@ -33,6 +35,24 @@ public AmazonEC2 getEcC2Client(Provider provider, Region r) {
       .build();
   }
 
+  @Override
+  public boolean isAWSValidCredentials(Map<String, String> config) {
+    BasicAWSCredentials awsCreds = new BasicAWSCredentials(config.get("AWS_ACCESS_KEY_ID"),
+      config.get("AWS_SECRET_ACCESS_KEY"));
+    try {
+      AmazonEC2 ec2Client = AmazonEC2ClientBuilder
+        .standard()
+        .withCredentials(new AWSStaticCredentialsProvider(awsCreds))
+        .withRegion(Regions.US_EAST_1)
```

Comment:
Why the region is hardcoded?

Reply:
Idea here is to validate the credentials by connecting to ec2. If we are able to connect it means credentials are valid else we will get some exceptions(Returning false for such scenarios)

Reply:
This particular region could be disabled for the customer's account (see description of describeRegions). So the check will fail, I suppose. But actually the credentials could be good for another region used by the customer.

Reply:
Yep. IIUC we should run against all the regions that have been selected in configuration for the provider and make sure creds work  for all. But need to understand how it all fits in the UI.  
i.e. we can validate config as a whole before saving it. Not just access key alone.

Reply:
If you remove withRegions it will use default region. That is better than explicitly selecting a region. 
But like I said ideally run dry run commands against all the regions in the provider config - may be that can happen in a separate PR.

Reply:
@sb-yb  I see, we don't have regions in the provider config(they are calling /bootstrap API to save these details. )
Running dry run commands against all the regions in the provider config may increase API response time(Performance issue)

Trying to perform some other operations without regions.

Reply:
Without specifying any regions, getting below error.
2021-01-12 00:18:35.966 ERROR AWSCloudImpl.java:56 [application-akka.actor.default-dispatcher-10] Unable to find a region via the region provider chain. Must provide an explicit region in the builder or setup environment to supply a region.

We needed regions in the backend.  UI has to send the regions details as part of POST /providers API payload. 
Please suggest!

Reply:
These are my suggestions:
1. Pick one region from provider/universe config (depending on context)
2. Basically the preflight check is holistic like I described above. 
3. Try to write code so as to reuse production client creation instead of doing something different.
4. Make DryRun request. Then you can also make mutate requests and get concrete permissions failure information that you can then expose to the user. So that they know whats wrong rather than "something is wrong" - without us having to look at the logs.